### ISSUE TITLE
Accessing managed memory while kernel is running will cause segmentation fault on devices without `concurrentManagedAccess` support

### ISSUE THREAD

**Issue by FredBill1** [ptal/turbo#16]
Hi, I was trying to excute `./build/gpu-release/turbo -s -v -i -t 20000 benchmarks/example_wordpress7_500.fzn` in the README and got a segmentation fault on this line (located with gdb): 

https://github.com/ptal/turbo/blob/9c439ff66f2ba1935915b9c4af7e1d4fd84ca2e2/include/gpu_solving.hpp#L143

which is called by this thread:

https://github.com/ptal/turbo/blob/9c439ff66f2ba1935915b9c4af7e1d4fd84ca2e2/include/gpu_solving.hpp#L572

I was using RTX2080super with nvcc12.1 on wsl2 ubuntu22.04.

I think it may be caused by this reason: [gpu-exclusive-access-to-managed-memory](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#gpu-exclusive-access-to-managed-memory)

> It is not permitted for the CPU to access any managed allocations or variables while the GPU is active for devices with `concurrentManagedAccess` property set to 0. On these systems concurrent CPU/GPU accesses, even to different managed memory allocations, will cause a segmentation fault because the page is considered inaccessible to the CPU.

As the following code prints a `0` on my environment:

```cpp
#include <iostream>
int main() {
    cudaDeviceProp prop;
    cudaGetDeviceProperties(&prop, 0);
    std::cout << prop.concurrentManagedAccess << std::endl;
}
```

**Comment by ptal** [comment]
Hi, thanks for reporting. I didn't add proper check of the GPU capabilities, will add that soon.
I had to use atomics to design a producer/consumer in order to print solutions on the CPU side because "printf" was too slow on the GPU for some problems (yeah crazy).

You can try out version 1.0.1, it might work on your GPU because the printing was on the GPU back then.

**Comment by FredBill1** [comment]
Thanks for the reply. I have tried version 1.0.1. It also generates segmentation error on this line:

https://github.com/ptal/turbo/blob/e37440f9cf01992f9f5e7725544249a04d26c9ff/include/common_solving.hpp#L59

which possibly due to the assignment of `a.stats.duration`, as it has been moved to managed memory with the struct `GridData` as well.

I removed `check_timeout(grid_data.root, start)` from this line and it stoped crashing.

https://github.com/ptal/turbo/blob/e37440f9cf01992f9f5e7725544249a04d26c9ff/include/gpu_solving.hpp#L495

**Comment by ptal** [comment]
I reopen to keep track of the `prop.concurrentManagedAccess` thing.

**Comment by ptal** [comment]
Fix in PR #18
